crates/vm2/src/callframe.rs

```diff
@@ -133,17 +135,29 @@ impl<T, W> Callframe<T, W> {
         })
     }
 
+    /// Gets a raw inferred program counter. This value can be garbage if the frame is on an invalid instruction or free panic.
+    #[allow(clippy::cast_possible_wrap)] // false positive: `Instruction` isn't that large
+    pub(crate) fn get_raw_pc(&self) -> isize {
+        // We cannot use `<*const _>::offset_from` because `self.pc` isn't guaranteed to be allocated within `self.program`
+        // (invalid instructions and free panics aren't).
+        let offset_in_bytes =
+            self.pc as isize - ptr::from_ref(self.program.instruction(0).unwrap()) as isize;
+        offset_in_bytes / mem::size_of::<Instruction<T, W>>() as isize
+    }
+
+    // FIXME: can overflow / underflow on invalid instruction / free panic
```

Comment:
Are there guarantees that this method in this case? I don't think this is the case; e.g., the host can create a snapshot after VM execution has panicked.

Reply:
I've added more details to the comment and "downgraded" it to TODO.